src/object/obj_tx.c

```diff
@@ -0,0 +1,1047 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * obj_tx: DAOS Transaction
+ *
+ * This module is part of libdaos. It implements the DAOS transaction API.
+ */
+#define D_LOGFAC	DD_FAC(object)
+
+#include <daos_task.h>
+#include <daos_types.h>
+#include <daos/common.h>
+#include <daos/object.h>
+#include <daos/container.h>
+#include <daos/pool.h>
+#include <daos/mgmt.h>
+#include <daos/dtx.h>
+#include <daos/task.h>
+#include "obj_rpc.h"
+#include "obj_internal.h"
+
+enum {
+	TX_OPEN,
+	TX_COMMITTING,
+	TX_COMMITTED,
+	TX_ABORTED,
+	TX_FAILED,
+};
+
+struct dc_tx_sub_req {
```

Comment:
would be good to add a comment here for each member.

Reply:
I will refresh the patch with more comment.